src/Microsoft.OpenApi.CSharpAnnotations.DocumentGeneration/Extensions/StringExtensions.cs

```diff
@@ -105,6 +105,30 @@ public static string RemoveBlankLines(this string value)
                 .Trim();
         }
 
+        /// <summary>
+        /// Removes the duplicate string from parameter name to work around the rosyln issue.
+        /// https://github.com/dotnet/roslyn/issues/26292.
+        /// e.g if the value is service-$Catalog-Id-1-$Catalog-Id-1, this will return
+        /// service-$Catalog-Id
+        /// </summary>
+        /// <param name="value">The param name to update.</param>
+        /// <returns>The updated value.</returns>
+        public static string RemoveDuplicateString(this string value)
+        {
+            if (string.IsNullOrEmpty(value))
+            {
+                return value;
+            }
+
+            // Find the first special character among -,$,@,_ in the string.
```

Comment:
Could you add reference to how we know these are the only characters affected? The issue linked above doesn't explicitly talk about it.

Reply:
i have just tried myself so hit and trial :-)

---
In reply to: [253636074](https://github.com/Microsoft/OpenAPI.NET.CSharpAnnotations/pull/198#discussion_r253636074) [](ancestors = 253636074)

Reply:
Also, let's add some comments to explain how duplication works with multiple special chars since it's not obvious.

Reply:
sure can do

---
In reply to: [253637576](https://github.com/Microsoft/OpenAPI.NET.CSharpAnnotations/pull/198#discussion_r253637576) [](ancestors = 253637576)